pkg/instancetype/apply/gpu_test.go

```diff
@@ -0,0 +1,56 @@
+//nolint:dupl
+package apply_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	virtv1 "kubevirt.io/api/core/v1"
+	v1beta1 "kubevirt.io/api/instancetype/v1beta1"
+
+	"kubevirt.io/kubevirt/pkg/instancetype/apply"
+	"kubevirt.io/kubevirt/pkg/libvmi"
+)
+
+var _ = Describe("Apply to vmi - gpu", func() {
+	var (
+		vmi            *virtv1.VirtualMachineInstance
+		preferenceSpec *v1beta1.VirtualMachinePreferenceSpec
+
+		vmiApplier       = apply.NewVMIApplier()
+		field            = k8sfield.NewPath("spec", "template", "spec")
+		instancetypeSpec = &v1beta1.VirtualMachineInstancetypeSpec{
+			GPUs: []virtv1.GPU{
+				{
+					Name:       "barfoo",
+					DeviceName: "vendor.com/gpu_name",
+				},
+			},
+		}
+	)
+
+	BeforeEach(func() {
+		vmi = libvmi.New()
+	})
+
+	It("should apply to VMI", func() {
+		Expect(vmiApplier.ApplyToVMI(field, instancetypeSpec, preferenceSpec, &vmi.Spec, &vmi.ObjectMeta)).To(Succeed())
```

Comment:
Just to clarify it for me: The returned conflicts implementes the Error interface and therefore the `Succeed` match can be applied?

Reply:
https://github.com/kubevirt/kubevirt/pull/13753#discussion_r1931856614
https://github.com/kubevirt/kubevirt/blob/main/pkg/instancetype/conflict/conflicts.go#L90